### ISSUE TITLE
Inconsistencies with desktop with regard to "type in the answer" card types

### ISSUE THREAD

**Issue by sudomain** [ankidroid/Anki-Android#6196]
Two small issues (called A and B) regarding typing card templates that use {{type:fieldname}} or {{type:cloze:fieldname}}:

A. While learning new cards, if a typing card is being shown right after the user reviewed a different typing card, the answer that was typed into the first card will sometimes be copied over as the answer to the second. This sporadically happens when the user doesn't type an answer on the second card. It happens even if the cards are of different note types.  I attempted to reproduce this on desktop by syncing and reviewing the same cards, but I wasn't able to reproduce it.

B. Assume a cloze model has a front and back template like:
`
{{cloze:Text 1}}

{{cloze:Text 2}}

{{type:cloze:Text 2}}
`

Create a note of this model with `Text 1` containing the cloze references c1 and c2 and `Text 2` containing the cloze references c3 and c4. The references c1 and c2 will only require recall while c3 and c4 will require typing. Anki and AnkiDroid differ in how they render a card using this template. AnkiDroid will show a text input box on all cards of this note, even c1 and c2. Anki will only show a text input box on c3 and c4

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
AnkiDroid Version = 2.11alpha4

Android Version = 7.0

ACRA UUID = 0f431910-fac3-4b52-ba37-64f328e5d239

###### Research
*Enter an [x] character to confirm the points below:*

- [ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ x ] I have searched for similar existing issues here and on the user forum
- [ x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
Yeah, template rendering is a bit of a mess at the moment. Upstream has updated this area and is now relying on a method we can't directly port unfortunately and our last few changes here in AnkiDroid have resulted in regressions.

Can you tell me if this happened in 2.9.x as well? Knowing how we differ between 2.10 and 2.9 AnkiDroid is more telling with regard to what the right action for us is at this point

**Comment by sudomain** [comment]
Sure thing. I installed the 2.9.6 parallel release and set my AnkiDroid dir to the same as my alpha version. I was able to reproduce both A and B. 

I learned more about issue A: It only happens when "Type answer into card" is disabled under advanced settings. Additionally, it seems to mostly affect the recall cards described in issue B (c1 and c2). Seems as though both issues stem from how AnkiDroid handle cloze templates with the type marker {{type:cloze:Text 2}}

**Comment by mikehardy** [comment]
So if I understand correctly this is not a regression from 2.9 to 2.10? If so then this will have no priority for me but I'll happily merge improvements to our Template parser if they make sense and don't stray too far from upstream (which is already straying from us in this area unfortunately)

**Comment by sudomain** [comment]
> So if I understand correctly this is not a regression from 2.9 to 2.10?
Correct. Issue B is a cosmetic inconsistency between Anki and AnkiDroid (textbox being shown when not needed). Issue A seems like it would be somewhat rare unless someone uses alot of cloze and type in the answer cards. I don't currently use alot of these cards, though I will soon. If it becomes sufficiently annoying, I will look into making a PR

**Comment by mikehardy** [comment]
whew, very happy to hear it's not a regression. With fingers crossed I stomped out the last one of those in 2.10.2 :sweat_smile:

**Comment by Arthur-Milchior** [comment]
Are there any rule to know whether it would be a good first issue ?
I assume it's just a if somewhere to change, no complex deep knowledge eequired. Except for finding the right if, sadly

**Comment by mikehardy** [comment]
@Arthur-Milchior it's just personal taste, if it seems like a potentially good first issue it can't really hurt to mark it as such. New contributors are always free to ask for help localizing the problem, or thoughts on a fix - we're very responsive to that

**Comment by pree-T** [comment]
May I work on this?

**Comment by david-allison-1** [comment]
Sure!

**Comment by dae** [comment]
Now uses backend to compare text, so should be resolved.
